integration/client_delete_serviceInstance_test.go

```diff
@@ -0,0 +1,198 @@
+package integration
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aerogear/mobile-cli/pkg/apis/servicecatalog/v1beta1"
+	"k8s.io/client-go/pkg/api/v1"
+	appsv1 "k8s.io/client-go/pkg/apis/apps/v1beta1"
+)
+
+const deleteServicetestPath = "deleteServiceInstanceTestData/"
+
+func TestDeleteServiceInstance(t *testing.T) {
+
+	fhSyncServer := &ProvisionServiceParams{
+		ServiceName: "fh-sync-server",
+		Namespace:   fmt.Sprintf("--namespace=%s", *namespace),
+		Params: []string{
+			"-p MONGODB_USER_NAME=fhsync",
+			"-p MONGODB_USER_PASSWORD=fhsyncpass",
+			"-p MONGODB_ADMIN_PASSWORD=pass",
+		},
+	}
+
+	createInstance(t, fhSyncServer)
+	fhSyncID, err := getInstanceID(fhSyncServer)
+
+	if err != nil {
+		t.Fatalf("failed to get fh-syncID: %v", err)
+	}
+
+	tests := []struct {
+		name          string
+		fixture       string
+		args          []string
+		validate      func(t *testing.T)
+		expectedError error
+	}{
+		{
+			name:          "missing arguments",
+			fixture:       "missing-args.golden",
+			args:          []string{"delete", "serviceinstance", "", "--namespace=" + *namespace},
+			expectedError: errors.New("exit status 1"),
+		},
+		{
+			name:          "Invalid instance ID",
+			fixture:       "invalid-instanceId.golden",
+			args:          []string{"delete", "serviceinstance", "fdhfgnfgnfg", "--namespace=" + *namespace},
+			expectedError: errors.New("exit status 1"),
+		},
+		{
+			name:          "Successful Delete",
+			args:          []string{"delete", "serviceinstance", fhSyncID, "--namespace=" + *namespace},
+			expectedError: nil,
+			validate: func(t *testing.T) {
+				time.Sleep(time.Second * 10)
+				instance, err := getInstance(fhSyncServer)
+				if err != nil {
+					t.Fatalf("unexpected error %v", err)
+				}
+				for _, condition := range instance.Status.Conditions {
```

Comment:
what happens here if the serviceinstance is gone after ten seconds? seems the test would fail even though the serviceinstance was successfully deleted? Does getInstance return an error if it fails to find the serviceinstance? 
We should prob do some kind of logic to allow for it being gone at this point.

Reply:
Hi @maleck13 the sleep was just a temporary solution while I figured out how to delay that test properly, but it would probably be better to read the code if it was moved down into the 'for_, condition loop below it...

The fh-sync-server would always be there for a while (even if its marked for delete) so the getInstance(fhSyncServer) would always work (and not have an error) unless there was an unexpected error...

So, the sleep is just giving it long enough for the instance to be marked for delete - so that when it hits the 'for loop' below it, the following criteria is met:

condition.Type == "Ready" && condition.Status != "False" 

Does that make sense?

Reply:
right so instead of a sleep you could do a watch similar to  https://github.com/aerogear/mobile-cli/blob/master/pkg/cmd/services.go#L370 and watch for a deleted event or check each time the object was modified

Reply:
OK, thanks @maleck13